File: src/core/validators/IntegerProperty/IntegerPropertyMustNotMatchACommonSimpleType.js
// @flow
import type SymbolTable from '../SymbolTable';
import { integerPropertyErrorRule, includeIntegerPropertyRule } from './IntegerPropertyValidationRule';
import { propertyMustNotMatchACommonSimpleType } from '../ValidationHelper';

// eslint-disable-next-line no-unused-vars
function failureMessage(ruleContext: any, symbolTable: SymbolTable): string {
  return `Integer property '${ruleContext.propertyName().getText()}' has the same name as a common decimal, integer, short or string.  If intentional, use a shared property instead.`;
}

const validationRule = integerPropertyErrorRule(propertyMustNotMatchACommonSimpleType, failureMessage);
export { validationRule as default };

export const includeRule = includeIntegerPropertyRule(validationRule);

File: src/core/validators/IntegerProperty/IntegerPropertyMustNotMatchACommonSimpleType.cs
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.IntegerProperty
{
    public class IntegerPropertyMustNotMatchACommonSimpleType : ValidationRuleBase<MetaEdGrammar.IntegerPropertyContext>
    {
        private readonly ISymbolTable _symbolTable;

        public IntegerPropertyMustNotMatchACommonSimpleType(ISymbolTable symbolTable)
        {
            _symbolTable = symbolTable;
        }

        public override bool IsValid(MetaEdGrammar.IntegerPropertyContext context)
        {
            var identifierToMatch = context.propertyName().GetText();
            var commonDecimalType = MetaEdGrammar.TokenName(MetaEdGrammar.COMMON_DECIMAL);
            var commonIntegerType = MetaEdGrammar.TokenName(MetaEdGrammar.COMMON_INTEGER);
            var commonShortType = MetaEdGrammar.TokenName(MetaEdGrammar.COMMON_SHORT);
            var commonStringType = MetaEdGrammar.TokenName(MetaEdGrammar.COMMON_STRING);

            return !(_symbolTable.IdentifierExists(commonDecimalType, identifierToMatch) ||
                     _symbolTable.IdentifierExists(commonIntegerType, identifierToMatch) ||
                     _symbolTable.IdentifierExists(commonShortType, identifierToMatch) ||
                     _symbolTable.IdentifierExists(commonStringType, identifierToMatch));
        }

        public override string GetFailureMessage(MetaEdGrammar.IntegerPropertyContext context)
        {
            return string.Format("Integer property '{0}' has the same name as a common decimal, integer, short or string.  If intentional, use a shared property instead.", context.propertyName().GetText());
        }
    }
}
